src/utils/db.ts

```diff
@@ -0,0 +1,67 @@
+import { Collection, MongoClient } from 'mongodb';
+import { encryptCredential, decryptCredential } from './../utils/crypto';
+import { Credential } from './../types';
+import dotenv from 'dotenv';
+dotenv.config();
+
+let client: MongoClient;
+
+export async function connectDb(): Promise<void> {
+  if (!process.env.DB_URI) throw new Error('No DB_URI in the .env found');
+  client = new MongoClient(process.env.DB_URI);
+  try {
+    await client.connect();
+  } catch (error) {
+    throw new Error(`Database connection failed: ${error}`);
+  }
+}
+
+export function getCollection<T>(collectionName: string): Collection<T> {
+  return client.db().collection<T>(collectionName);
+}
+
+export function getCredentialCollection(): Collection<Credential> {
+  return getCollection<Credential>('credentials');
+}
+
+export async function addOrUpdateCredential(
+  credential: Credential,
+  key: string
+): Promise<void> {
+  const encryptedCredential = encryptCredential(credential, key);
+  await getCredentialCollection().updateMany(
+    { service: credential.service },
+    { $set: encryptedCredential },
+    { upsert: true }
+  );
+}
```

Comment:
It feels like all these functions (this one and the following ones) are a bit much to live inside this file. I wouldn't count them towards utilities, but rather one of the main functionalities of your backend